pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -836,6 +840,31 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
     }
   }
 
+  @VisibleForTesting
+  static String addRoutingPolicyInErrMsg(String errorMessage, String realtimeRoutingPolicy,
+      String offlineRoutingPolicy) {
+    if (realtimeRoutingPolicy != null && offlineRoutingPolicy != null) {
+      return String.format("%s, with routing policy: %s [realtime], %s [offline]", errorMessage, realtimeRoutingPolicy,
+          offlineRoutingPolicy);
+    }
+    if (realtimeRoutingPolicy != null) {
+      return String.format("%s, with routing policy: %s [realtime]", errorMessage, realtimeRoutingPolicy);
+    }
+    if (offlineRoutingPolicy != null) {
+      return String.format("%s, with routing policy: %s [offline]", errorMessage, offlineRoutingPolicy);
+    }
```

Comment:
Is there a reason to use `String.format` here? It is pretty inefficient and IMHO more difficult to read that just concatenating Strings.

Reply:
got it. it's error handling path, but anyway, lemme update to use +